product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/ScenarioTestBase.java

```diff
@@ -441,4 +446,67 @@ public boolean isTagsResponseContainsTag(JSONObject tagsResponse, String tagName
         }
         return false;
     }
+
+    /**
+     * Design, Implement and Manage (and publish if required) an API via the approach that the Publisher UI is
+     * performing, by calling required Jaggery APIs. This is to be used as a common method to develop/publish APIs for
+     * scenario tests.
+     *
+     * @param apiPublisher APIPublisherRestClient
+     * @param swaggerFileRelativePath Swagger file relative path to create API
+     * @param apiDeveloperUsername Developer name
+     * @param backendEndPoint Backendpoint url
+     * @param publish Whether to publish APi or not
+     * @param storeVisibility Store visibility level of the API
+     * @throws Exception
+     */
+    public void developSampleAPI(APIPublisherRestClient apiPublisher, String swaggerFileRelativePath,
```

Comment:
I'm wondering why this method should be in here instead of APIPublisherRestClient?

Reply:
We have defined APIPublisherRestClient to add methods to directly call particular Publisher Jaggery REST APIs. But this developSampleAPI(..) method is different right? It makes multiple REST calls as well do other operations to achieve the objective of developing a sample API. Methods currently existing within ScenarioTestBase are doing such that various operations, and they are using multiple Rest clients.

Reply:
IMO since it is a Rest client functionality more than a utility it will be better if we can have it in RestClient side

Reply:
Changed with cd65577